dubbo-metadata-report/dubbo-metadata-report-redis/src/main/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReport.java

```diff
@@ -37,7 +38,7 @@
 
     public RedisMetadataReport(URL url) {
         super(url);
-        pool = new JedisPool(new JedisPoolConfig(), url.getHost(), url.getPort());
+        pool = new JedisPool(new JedisPoolConfig(), url.getHost(), url.getPort(), Protocol.DEFAULT_TIMEOUT, url.getPassword());
```

Comment:
You can use `url.getParameter(Constants.TIMEOUT_KEY, Constants.DEFAULT_TIMEOUT)`. Get the timeout from url, if not present, use the default value.

Reply:
Right.